lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -722,3 +729,62 @@ hex = convertToBase Base16
 -- | Decode a 'ByteString' from base16
 fromHex :: ByteArray bout => ByteString -> Either String bout
 fromHex = convertFromBase Base16
+
+data ErrUnXPrvStripPub
+    = ErrNoRoundtripMismatch
+      -- ^ The resulting bytestring would have been unable to roundtrip using
+      -- @xPrvFromStrippedPubXPrv@. Most likely because the input @XPrv@ was
+      -- encrypted.
+    | ErrUnexpectedRoundtripError
+      -- ^ Something unexpected went wrong.
+    deriving (Eq, Show)
+
+-- | Extracts the raw xprv bytes as @private key <> chain code@, i.e. without
+-- the public key.
+--
+-- Returns @Left@ if the resulting bytestring fails to roundtrip. Most likely
+-- because the provided key was encrypted.
+unXPrvStripPub :: XPrv -> Either ErrUnXPrvStripPub ByteString
+unXPrvStripPub k = do
+    let res = stripPub . unXPrv $ k
+
+    -- Check that it roundtrips.
+    case (fmap unXPrv . xPrvFromStrippedPubXPrv $ res) of
+            Right bytes
+                | bytes == unXPrv k -> Right res
+                | otherwise         -> Left ErrNoRoundtripMismatch
+            Left  _                 -> Left ErrUnexpectedRoundtripError
+  where
+    -- Converts  prv <> pub <> cc
+    -- To        prv <>        cc
+    stripPub :: ByteString -> ByteString
+    stripPub xprv' = prv <> chainCode
+      where
+        (prv, rest) = BS.splitAt 64 xprv'
+        (_pub, chainCode) = BS.splitAt 32 rest
+
+data ErrXPrvFromStrippedPubXPrv
+    = ErrInputLengthMismatch Int Int -- ^ Expected, Actual
+    | ErrInternalError String
+    deriving (Eq, Show)
+
+-- Create a @XPrv@ from a prv <> cc bytestring.
+xPrvFromStrippedPubXPrv :: ByteString -> Either ErrXPrvFromStrippedPubXPrv XPrv
+xPrvFromStrippedPubXPrv x =
+    let
+        res = toXPrv $ CC.encryptedCreateDirectWithoutTweak prv cc pass
+    in
+        if BS.length x == expectedInputLength
+        then res
+        else Left $ ErrInputLengthMismatch expectedInputLength (BS.length x)
+  where
+    (prv, cc) = BS.splitAt 64 x
+    pass :: ByteString
+    pass = ""
+
+    expectedInputLength = 96
```

Comment:
Why 96? 

(I know why, but another reader might not).

Reply:
I elaborated on the format in the doc-comments of `unXPrvStripPub` and `xPrvFromStrippedPubXPrv`, and mentioned "96 bytes" clearly. I won't add a comment here, specifically.